### PR TITLE
[ISSUE-#4166] Optimize notifyCenter code

### PR DESCRIPTION
**Pull request by horizonzy**
## What is the purpose of the change
For-#4166

### CODE REVIEW

File: common/src/main/java/com/alibaba/nacos/common/notify/NotifyCenter.java [hunk 1]
```diff
@@ -250,13 +250,12 @@ private static void addSubscriber(final Subscriber consumer, Class<? extends Eve
     private static boolean removeSubscriber(final Subscriber consumer, Class<? extends Event> subscribeType) {
         
         final String topic = ClassUtils.getCanonicalName(subscribeType);
-        if (INSTANCE.publisherMap.containsKey(topic)) {
-            EventPublisher publisher = INSTANCE.publisherMap.get(topic);
-            publisher.removeSubscriber(consumer);
-            return true;
+        EventPublisher eventPublisher = INSTANCE.publisherMap.get(topic);
```
Comment (KomachiSion) [inline common/src/main/java/com/alibaba/nacos/common/notify/NotifyCenter.java:253]: I think the old implementation is better for readablilty

File: common/src/main/java/com/alibaba/nacos/common/notify/NotifyCenter.java [hunk 2]
```diff
@@ -281,18 +280,19 @@ public static boolean publishEvent(final Event event) {
...
             return INSTANCE.sharePublisher.publish(event);
         }
         
-        if (INSTANCE.publisherMap.containsKey(topic)) {
-            EventPublisher publisher = INSTANCE.publisherMap.get(topic);
-            return publisher.publish(event);
+        final String topic = ClassUtils.getCanonicalName(eventType);
+        
+        EventPublisher publisher = INSTANCE.publisherMap.get(topic);
+        if (publisher == null) {
```
Comment (KomachiSion) [inline common/src/main/java/com/alibaba/nacos/common/notify/NotifyCenter.java:290]: Same as above

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/alibaba/nacos/common/notify/NotifyCenter.java (hunk 1 above)
Comment (horizonzy) [reply common/src/main/java/com/alibaba/nacos/common/notify/NotifyCenter.java:253]: I think containsKey() and get() is duplicated. get() can cover containsKey(); Maybe should judge eventPublish != null?

Comment (KomachiSion) [reply common/src/main/java/com/alibaba/nacos/common/notify/NotifyCenter.java:253]: OK

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**